File: app/src/main/java/com/kisssum/smartcity/adapter/home/HotServiceListAdpater.java
package com.kisssum.smartcity.adapter.home;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.kisssum.smartcity.R;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class HotServiceListAdpater extends RecyclerView.Adapter<HotServiceListAdpater.DefaultViewMode> {
    List<Map<String, Object>> data;
    private Context context;

    public HotServiceListAdpater(Context context) {
        data = getData();
        this.context = context;
    }

    @NonNull
    @Override
    public DefaultViewMode onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_style_hot_service, parent, false);
        return new DefaultViewMode(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DefaultViewMode holder, int position) {
        holder.img.setImageResource((Integer) data.get(position).get("img"));
        holder.name.setText(data.get(position).get("title").toString());

        holder.itemView.setOnClickListener(v -> {
            BottomNavigationView view = ((Activity) context).findViewById(R.id.bottomNavigationView);
            view.setSelectedItemId(R.id.item_allservice);
        });
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public List<Map<String, Object>> getData() {
        ArrayList list = new ArrayList();
        Map<String, Object> map;
        int[] imgs = new int[]{
                R.drawable.top_view_pager_1,
                R.drawable.top_view_pager_2,
                R.drawable.top_view_pager_3,
                R.drawable.top_view_pager_4,
                R.drawable.top_view_pager_5,
                R.drawable.top_view_pager_1
        };

        for (int img : imgs) {
            map = new HashMap<>();
            map.put("img", img);
            map.put("title", "热门主题");
            list.add(map);
        }
        return list;
    }

    class DefaultViewMode extends RecyclerView.ViewHolder {
        ImageView img;
        TextView name;

        public DefaultViewMode(@NonNull View itemView) {
            super(itemView);

            img = itemView.findViewById(R.id.hotImg);
            name = itemView.findViewById(R.id.hotName);
        }
    }
}

File: app/src/main/java/com/kisssum/smartcity/adapter/home/HotServiceListAdpater.kt
package com.kisssum.smartcity.adapter.home

import android.app.Activity
import android.content.Context
import android.graphics.Bitmap
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.android.volley.toolbox.ImageRequest
import com.android.volley.toolbox.Volley
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.kisssum.smartcity.R
import com.kisssum.smartcity.adapter.home.HotServiceListAdpater.DefaultViewMode
import com.kisssum.smartcity.tool.API
import java.util.*
import kotlin.collections.ArrayList

class HotServiceListAdpater(val context: Context, val data: ArrayList<Map<String, Any>>) : RecyclerView.Adapter<DefaultViewMode>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DefaultViewMode {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.list_style_hot_service, parent, false)
        return DefaultViewMode(view)
    }

    override fun onBindViewHolder(holder: DefaultViewMode, position: Int) {
        Volley.newRequestQueue(context).apply {
            val imageRequest = ImageRequest(
                    API.getBaseUrl(context) + data[position]["imgUrl"].toString(),
                    {
                        holder.img.setImageBitmap(it)
                    },
                    0, 0, Bitmap.Config.RGB_565,
                    {}
            )

            this.add(imageRequest)
        }

        holder.name.text = data[position]["title"].toString()
        holder.itemView.setOnClickListener { v: View? ->
            val view: BottomNavigationView = (context as Activity).findViewById(R.id.bottomNavigationView)
            view.selectedItemId = R.id.item_allservice
        }
    }

    override fun getItemCount(): Int {
        return data.size
    }

    inner class DefaultViewMode(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val img = itemView.findViewById<ImageView>(R.id.hotImg)
        val name = itemView.findViewById<TextView>(R.id.hotName)
    }
}